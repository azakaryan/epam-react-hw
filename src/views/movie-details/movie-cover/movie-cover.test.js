import React from 'react';
import { shallow } from 'enzyme';
import expectEnzymify from 'expect';
import enzymify from 'expect-enzyme';
import MovieCover from './movie-cover';

expectEnzymify.extend(enzymify());
const movie = {
  budget: 160000000,
  genres: ['Action', 'Thriller', 'Science Fiction', 'Mystery', 'Adventure'],
  id: 27205,
  overview:
    "Cobb, a skilled thief who commits corporate espionage by infiltrating the subconscious of his targets is offered a chance to regain his old life as payment for a task considered to be impossible: 'inception', the implantation of another person's idea into a target's subconscious.",
  poster_path:
    'https://image.tmdb.org/t/p/w500/qmDpIHrmpJINaRKAfWQfftjCdyi.jpg',
  release_date: '2010-07-14',
  revenue: 825532764,
  runtime: 148,
  tagline: 'Your mind is the scene of the crime.',
  title: 'Inception',
  vote_average: 8.2,
  vote_count: 16798,
};

describe('Movie Cover Component', () => {
  it('should render correctly', () => {
    const component = shallow(<MovieCover movie={movie} />);
    expect(component).toMatchSnapshot();
  });

  it('renders a Movie Cover component', () => {
    const wrapper = shallow(<MovieCover movie={movie} />);
    expectEnzymify(wrapper).toBeA('div');
  });

  it('renders empty component', () => {
    const wrapper = shallow(<MovieCover />);
    expect(wrapper).toEqual({});
  });
});
