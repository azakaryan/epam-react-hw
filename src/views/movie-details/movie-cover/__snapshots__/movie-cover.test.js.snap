// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Cover Component should render correctly 1`] = `
<div
  className="main"
>
  <div>
    <div
      className="left_section"
    >
      <img
        alt="speakom"
        src="https://image.tmdb.org/t/p/w500/qmDpIHrmpJINaRKAfWQfftjCdyi.jpg"
      />
    </div>
    <div
      className="right_section"
    >
      <div>
        <div
          className="title"
        >
          Inception
        </div>
        <div
          className="rating"
        >
          16798
        </div>
      </div>
      <div>
        Oscar-winning Movies
      </div>
      <div>
        <div
          className="year"
        >
          2010
        </div>
        <div
          className="duration"
        >
          154 min
        </div>
      </div>
      <div>
        Cobb, a skilled thief who commits corporate espionage by infiltrating the subconscious of his targets is offered a chance to regain his old life as payment for a task considered to be impossible: 'inception', the implantation of another person's idea into a target's subconscious.
      </div>
    </div>
  </div>
  <div
    className="genres"
  >
    Action & Thriller & Science Fiction & Mystery & Adventure
  </div>
</div>
`;
